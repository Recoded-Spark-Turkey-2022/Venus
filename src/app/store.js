// eslint-disable-next-line no-unused-vars
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import listingReducer from '../features/listings/listingSlice';

export const store = configureStore({
  reducer: {
    listing: listingReducer,
  },
  // eslint-disable-next-line no-shadow
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});
