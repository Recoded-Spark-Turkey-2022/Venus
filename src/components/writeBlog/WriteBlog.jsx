import { useState } from "react";

const WriteBlogC=()=>{

    const[title, setTitle]=useState('');
    const[blogText, setBlogText]=useState('');
    const[banner, setBanner]=useState(null);

    const submitBlog= (e)=>{
        e.preventDefault();
 // Submit banner img, title, blog text with auth
    }
    const handleChangeBanner=(e)=>{
        console.log(e.target.files[0]);
        setBanner(e.target.files[0]);
    }

    return(
        <div className='w-full'>
        <form name="contact" className='flex flex-col justify-center md:justify-start mt-5 ' onSubmit={submitBlog}>
        <div className="image-div flex flex-col justify-center md:justify-start mt-5 gap-2 my-2 text-darkGrey">
          { /* eslint-disable-next-line jsx-a11y/label-has-associated-control */ }
          <label className="" htmlFor="image">Choose a banner image:</label>
          <input 
          type="file" 
          id="image" 
          name="image" 
          accept="image/png, image/jpeg" 
          required
          onChange={handleChangeBanner}/>
          </div>
          <div className='input-div flex flex-col justify-center md:justify-start mt-5 gap-2 my-2 '>
          { /* eslint-disable-next-line jsx-a11y/label-has-associated-control */ }
            <label className="text-darkGrey">
                Title:
            </label>
            <input 
            placeholder='Write Title' 
            required 
            onChange={(e)=> setTitle(e.target.value)} 
            className="border-1 p-2 rounded-2xl border border-darkGrey"/>
            </div>
            <div className='blog-div flex flex-col justify-center md:justify-start mt-5 gap-2 my-2'>
            { /* eslint-disable-next-line jsx-a11y/label-has-associated-control */ }
            <label className="text-darkGrey">
                Blog:
            </label>
            <textarea 
            placeholder='Write Blog' 
            required 
            onChange={(e)=> setBlogText(e.target.value)}
            className='p-2 rounded-full border border-1 border-darkGrey'/>
            </div>
            <button onClick={submitBlog} id='mediumBlue-button' className='w-32 rounded-full mt-2 whitespace-nowrap py-2' type='submit'>Submit Blog</button>
            </form>
            </div>
    )
}
export default WriteBlogC;
