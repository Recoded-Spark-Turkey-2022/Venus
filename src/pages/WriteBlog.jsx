import Container from "../components/UI/Container";
import WriteBlogC from "../components/writeBlog/WriteBlog";
import writeBlog from '../assets/writeBlog.png';



const WriteBlog=()=> {

   

    return(
        <div id='' className=" mt-10 bg-inlightBlue">
        <Container>
         <div className="grid grid-cols-1 md:grid-cols-2  gap-6 pt-12 md:p-20 pb-5">
          <div className="grid justify-items-center md:justify-items-start">
           <h1 className="text-darkBlue text-center md:text-left text-5xl font-bold mb-5">
             Write a Blog
           </h1>
          <div className="grid justify-items-start md:justify-items-center w-full md:w-2/3">
          <WriteBlogC/>
          </div>
          </div>
          <div className="flex justify-center items-center row-end-1 md:col-end-3">
           <img src={writeBlog} alt="Bannerimage"/>
          </div>
          </div>
        
        </Container>
        </div>
        
       

    )
}

export default WriteBlog;