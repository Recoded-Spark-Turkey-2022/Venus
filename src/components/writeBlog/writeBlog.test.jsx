import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { store } from '../../app/store';
import WriteBlog from './WriteBlog';

jest.mock('react-i18next', () => ({
    useTranslation: () => {
      return {
        t: (str) => str
      };
    },
  }));

describe('WriteBlog', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(
          <Router>
            <Provider store={store}>
              <WriteBlog />
            </Provider>
          </Router>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });