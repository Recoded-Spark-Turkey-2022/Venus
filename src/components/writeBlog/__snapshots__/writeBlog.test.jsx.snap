// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WriteBlog renders correctly 1`] = `
<div
  className="w-full"
>
  <form
    className="flex flex-col justify-center md:justify-start "
    name="contact"
    onSubmit={[Function]}
  >
    <div
      className="image-div flex flex-col  gap-2 my-2 text-darkGrey"
    >
      <label
        className=""
        htmlFor="imageUrl"
      >
        Blog.ChooseImage
      </label>
      <input
        accept="image/png, image/jpeg"
        className="bg-white w-full px-3 py-2.5"
        id="ImageUrl"
        max="1"
        onChange={[Function]}
        required={true}
        type="file"
      />
    </div>
    <div
      className="input-div flex flex-col  mt-5 gap-2 my-2 "
    >
      <label
        className="text-darkGrey"
      >
        Blog.Title
      </label>
      <input
        className="border-1  w-full border border-darkGrey"
        id="title"
        onChange={[Function]}
        placeholder="Blog.TitlePlaceholder"
        required={true}
        type=""
      />
    </div>
    <div
      className="input-div flex flex-col  mt-5 gap-2 my-2 "
    >
      <label
        className="text-darkGrey"
      >
        Blog.Preview
      </label>
      <input
        className="border-1 border w-full border-darkGrey"
        id="text"
        onChange={[Function]}
        placeholder="Blog.PreviewPlaceholder"
        required={true}
        type=""
      />
    </div>
    <div
      className="blog-div flex flex-col  mt-5 gap-2 my-2"
    >
      <label
        className="text-darkGrey"
      >
        Blog.Blog
      </label>
      <textarea
        className="rounded-full border max-h-[300px] border-1 border-darkGrey"
        id="content"
        onChange={[Function]}
        placeholder="Blog.BlogPlaceholder"
        required={true}
      />
    </div>
    <div
      className="flex justify-center md:justify-between items-center mt-2"
    >
      <button
        className="w-32 rounded-full mt-2 whitespace-nowrap py-2"
        id="mediumBlue-button"
        onClick={[Function]}
        type="submit"
      >
        Blog.SubmitBlog
      </button>
      <button
        className="w-32 ml-4 rounded-full mt-2 whitespace-nowrap py-2"
        id="mediumBlue-button"
        onClick={[Function]}
        type="button"
      >
        Go Back
      </button>
    </div>
  </form>
  <div
    className="Toastify"
  />
</div>
`;
